### PR TITLE
🔧  QoL Configuration

### PR DESCRIPTION
**Pull request by Gary-H9**
This PR relates to [this](https://github.com/ministryofjustice/analytical-platform/issues/4613) PR. It adds: 

- Numerous GitHub Actions
  - Code Scanning, Linting, build, release etc 
- Ministry of Justice Development Containers
  - including a `postgresql` container in Docker in Docker 
  - numerous tools to aid development
- configures the application to use a `postgresql` database
- `.editorconfig`, `Makefile`, `.flake8`, `.pre-commit-config.yaml`
- `package.json` + `package-lock.json` - these will be consumed later 
- `requirements.txt` + `requirements.dev.txt`

The following files are created with the knowledge that they will be edited/superseded in short order.

- README 
- `Dockerfile` 
- `chart` directory

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "analytical-platform-ollamate",
+  "image": "ghcr.io/ministryofjustice/devcontainer-base:latest",
+  "features": {
+    "ghcr.io/devcontainers/features/node:1": {
+      "version": "20.11.1"
```
Comment (jacobwoffenden) [inline .devcontainer/devcontainer.json]:
```suggestion
      "version": "20.15.1"
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,3 +1 @@
-# Add a team or username to this file
-# Example:
-# *   @ministryofjustice/operations-engineering
+@ministryofjustice/analytical-platforms
```
Comment (jacobwoffenden) [inline .github/CODEOWNERS]:
```suggestion
@ministryofjustice/analytical-platform
```

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        id: checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Build Image
+        id: build_image
+        uses: docker/build-push-action@4a13e500e55cf31b7a5d59a38ab2040ab0f42f56 # v5.1.0
+        with:
+          push: false
+          load: true
+          tags: dashboard
```
Comment (jacobwoffenden) [inline .github/workflows/build-test.yml]:
```suggestion
          tags: ollamate
```

File: .github/workflows/dependency-review.yml
Comment (jacobwoffenden) [inline .github/workflows/dependency-review.yml:1]: this is rolling back versions

File: .github/workflows/release.yml
Comment (jacobwoffenden) [inline .github/workflows/release.yml:1]: Indentation is wrong

File: .github/workflows/scan-image.yml
Comment (jacobwoffenden) [inline .github/workflows/scan-image.yml:1]: Indentation is wrong

File: README.md
Comment (jacobwoffenden) [inline README.md:1]: `template-repository` needs replacing in standards URL

**Review by jacobwoffenden — DISMISSED** [review]

File: .github/workflows/scan-image.yml
Comment (Gary-H9) [reply .github/workflows/scan-image.yml:1]: Resolved

File: .github/workflows/release.yml
Comment (Gary-H9) [reply .github/workflows/release.yml:1]: Resolved

File: .github/workflows/dependency-review.yml
Comment (Gary-H9) [reply .github/workflows/dependency-review.yml:1]: Resolved

**Review by jacobwoffenden — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      contents: read
+      security-events: write
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["javascript", "python"]
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
Comment (jacobwoffenden) [inline .github/workflows/codeql-analysis.yml]: 4.1.7 is available now

File: README.md [hunk 5]
```diff
@@ -1,66 +1,31 @@
...
...W1UAMI3o5CD4bmuOlwHaC6xakgZ4Z+ibgSxnOgcAI4uavI27jEII7909dL5VSrimlPKgeQ6TJCZVQjwaOLaW8BfyWbPEa1SaiTH1VfSENd85NDxHt1plA71LKRvX4BDaAKFlTgLeALtliDUqPrSV6SQCBlypgFlbmIIrCDcAl6nPAawmYhlLKFuB6IrkXAadUNj6TXlhDcCNEB/Jn4FcE0f4UWEl0NyWNvZxGTs89z6ZnatIIrCdqcCtRJmcCPwCeSN3N1Iu6T4VaFhm9n+riypouBnepLsk9p6p35fzwvDSX5eVQvaDOzjnqzTl+1KC53+XzLINHd65O6lD1DnWbepPBhQ3q2jQyW+2oDkkAtdt5udpb7W+Q/OFGA7ol1zxu1tc8zNHqXercfDfQIOZm9fR815Cpt5PnVqsr1F51wI9QnzU63xZ1o/rdPPmt6enV6sXqHPVqdXOCe1rtrg5W7zNI+m712Ir+cer4POiqfHeJSVe1Raemwnm7xD3mD1E/Z3wIjcsTdlZnqO8bFeNB9c30zgVG2euYa69QJ+9G90lG+99bfdIoo5PU4w362xHePxl1slMab6tV72KUxDvzlAMT8G0ZohXq39VX1bNzzxij9K1Qb9lhdGe931B/kR6/zCwY9YvuytCsMlj+gbr5SemhqkyuzE8xau4MP865JvWNuj0b1YuqDkgvH2GkURfakly01Cg7Cw0+qyXxkjojq9Lw+vT2AUY+DlF/otYq1Ixc35re2V7R8aTRg2KUv7+ou3x/14PsUBn3NG51S0XpG0Z9PcOPKWSS0SKNUo9Rv2Mmt/G5WpPF6pHGra7Jv410OVsdaz217AbkAPX3ubkm240belCuudT4Rp5p/DyC2lf9mfq1iq5eFe8/lu+K0YrVp0uret4nAkwlB6vzjI/1PxrlrTp/oNHbzTJI92T1qAT+BfW49MhMg6JUp7ehY5a6Tl2jjmVvitF9fxo5Yq8CaAfAkzLMnySt6uz/1k6bPx59CpCNxGfoSKA30IPoH7cQXdArwCOllFX/i53P5P9a/gNkKpsCMFRuFAAAAABJRU5ErkJggg==)](https://operations-engineering-reports.cloud-platform.service.justice.gov.uk/public-report/analytical-platform-ollamate)
```
Comment (jacobwoffenden) [inline README.md]:
```suggestion
[![repo standards badge](https://img.shields.io/endpoint?labelColor=231f20&color=005ea5&style=for-the-badge&label=MoJ%20Compliant&url=https%3A%2F%2Foperations-engineering-reports.cloud-platform.service.justice.gov.uk%2Fapi%2Fv1%2Fcompliant_public_repositories%2Fendpoint%2Fanalytical-platform-ollamate&logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAoCAYAAACM/rhtAAAABmJLR0QA/wD/AP+gvaeTAAAHJElEQVRYhe2YeYyW1RWHnzuMCzCIglBQlhSV2gICKlHiUhVBEAsxGqmVxCUUIV1i61YxadEoal1SWttUaKJNWrQUsRRc6tLGNlCXWGyoUkCJ4uCCSCOiwlTm6R/nfPjyMeDY8lfjSSZz3/fee87vnnPu75z3g8/kM2mfqMPVH6mf35t6G/ZgcJ/836Gdug4FjgO67UFn70+FDmjcw9xZaiegWX29lLLmE3QV4Glg8x7WbFfHlFIebS/ANj2oDgX+CXwA9AMubmPNvuqX1SnqKGAT0BFoVE9UL1RH7nSCUjYAL6rntBdg2Q3AgcAo4HDgXeBAoC+wrZQyWS3AWcDSUsomtSswEtgXaAGWlVI2q32BI0spj9XpPww4EVic88vaC7iq5Hz1BvVf6v3qe+rb6ji1p3pWrmtQG9VD1Jn5br+Knmm70T9MfUh9JaPQZu7uLsR9gEsJb3QF9gOagO7AuUTom1LpCcAkoCcwQj0VmJregzaipA4GphNe7w/MBearB7QLYCmlGdiWSm4CfplTHwBDgPHAFmB+Ah8N9AE6EGkxHLhaHU2kRhXc+cByYCqROs05NQq4oR7Lnm5xE9AL+GYC2gZ0Jmjk8VLKO+pE4HvAyYRnOwOH5N7NhMd/WKf3beApYBWwAdgHuCLn+tatbRtgJv1awhtd838LEeq30/A7wN+AwcBt+bwpD9AdOAkYVkpZXtVdSnlc7QI8BlwOXFmZ3oXkdxfidwmPrQXeA+4GuuT08QSdALxC3OYNhBe/TtzON4EziZBXD36o+q082BxgQuqvyYL6wtBY2TyEyJ2DgAXAzcC1+Xxw3RlGqiuJ6vE6QS9VGZ/7H02DDwAvELTyMDAxbfQBvggMAAYR9LR9J2cluH7AmnzuBowFFhLJ/wi7yiJgGXBLPq8A7idy9kPgvAQPcC9wERHSVcDtCfYj4E7gr8BRqWMjcXmeB+4tpbyG2kG9Sl2tPqF2Uick8B+7szyfvDhR3Z7vvq/2yqpynnqNeoY6v7LvevUU9QN1fZ3OTeppWZmeyzRoVu+rhbaHOledmoQ7LRd3SzBVeUo9Wf1DPs9X90/jX8m/e9Rn1Mnqi7nuXXW5+rK6oU7n64mjszovxyvVh9WeDcTVnl5KmQNcCMwvpbQA1xE8VZXhwDXAz4FWIkfnAlcBAwl6+SjD2wTcmPtagZnAEuA3dTp7qyNKKe8DW9UeBCeuBsbsWKVOUPvn+MRKCLeq16lXqLPVFvXb6r25dlaGdUx6cITaJ8fnpo5WI4Wuzcjcqn5Y8eI/1F+n3XvUA1N3v4ZamIEtpZRX1Y6Z/DUK2g84GrgHuDqTehpBCYend94jbnJ34DDgNGArQT9bict3Y3p1ZCnlSoLQb0sbgwjCXpY2blc7llLW1UAMI3o5CD4bmuOlwHaC6xakgZ4Z+ibgSxnOgcAI4uavI27jEII7909dL5VSrimlPKgeQ6TJCZVQjwaOLaW8BfyWbPEa1SaiTH1VfSENd85NDxHt1plA71LKRvX4BDaAKFlTgLeALtliDUqPrSV6SQCBlypgFlbmIIrCDcAl6nPAawmYhlLKFuB6IrkXAadUNj6TXlhDcCNEB/Jn4FcE0f4UWEl0NyWNvZxGTs89z6ZnatIIrCdqcCtRJmcCPwCeSN3N1Iu6T4VaFhm9n+riypouBnepLsk9p6p35fzwvDSX5eVQvaDOzjnqzTl+1KC53+XzLINHd65O6lD1DnWbepPBhQ3q2jQyW+2oDkkAtdt5udpb7W+Q/OFGA7ol1zxu1tc8zNHqXercfDfQIOZm9fR815Cpt5PnVqsr1F51wI9QnzU63xZ1o/rdPPmt6enV6sXqHPVqdXOCe1rtrg5W7zNI+m712Ir+cer4POiqfHeJSVe1Raemwnm7xD3mD1E/Z3wIjcsTdlZnqO8bFeNB9c30zgVG2euYa69QJ+9G90lG+99bfdIoo5PU4w362xHePxl1slMab6tV72KUxDvzlAMT8G0ZohXq39VX1bNzzxij9K1Qb9lhdGe931B/kR6/zCwY9YvuytCsMlj+gbr5SemhqkyuzE8xau4MP865JvWNuj0b1YuqDkgvH2GkURfakly01Cg7Cw0+qyXxkjojq9Lw+vT2AUY+DlF/otYq1Ixc35re2V7R8aTRg2KUv7+ou3x/14PsUBn3NG51S0XpG0Z9PcOPKWSS0SKNUo9Rv2Mmt/G5WpPF6pHGra7Jv410OVsdaz217AbkAPX3ubkm240belCuudT4Rp5p/DyC2lf9mfq1iq5eFe8/lu+K0YrVp0uret4nAkwlB6vzjI/1PxrlrTp/oNHbzTJI92T1qAT+BfW49MhMg6JUp7ehY5a6Tl2jjmVvitF9fxo5Yq8CaAfAkzLMnySt6uz/1k6bPx59CpCNxGfoSKA30IPoH7cQXdArwCOllFX/i53P5P9a/gNkKpsCMFRuFAAAAABJRU5ErkJggg==)](https://operations-engineering-reports.cloud-platform.service.justice.gov.uk/public-report/analytical-platform-ollamate)
```

File: chart/templates/.helpers.tpl
Comment (jacobwoffenden) [inline chart/templates/.helpers.tpl:1]: Replace references to `actions-runner` in the chart

File: chart/values.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+---
+nameOverride: ""
+fullnameOverride: ""
+
+replicaCount: 1
+
+image:
+  pullPolicy: IfNotPresent
+  repository: ghcr.io/ministryofjustice/analytical-platform-ollamate
+  tag: 0.0.1
```
Comment (jacobwoffenden) [inline chart/values.yaml]:
```suggestion
  tag: ""
```
Set to blank to use `appVersion`

File: chart/values.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  runAsNonRoot: true
+  runAsUser: 10000
+
+resources:
+  requests:
+    cpu: 1
+    memory: "5Gi"
+  limits:
+    cpu: 2
+    memory: "7Gi"
```
Comment (jacobwoffenden) [inline chart/values.yaml]:
```suggestion
resources:
  requests:
    cpu: 1
    memory: "5Gi"
  limits:
    cpu: 2
    memory: "7Gi"
```
These probably want tuning

File: chart/values.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+github:
+  organisation:
+  repository:
+  token:
+  runner:
+    labels:
```
Comment (jacobwoffenden) [inline chart/values.yaml]: This can go

File: chart/values.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+ephemeral:
+  enabled: false
+  karpenter:
+    enabled: false
+    nodePool: "general-on-demand"
+  keda:
+    maxReplicaCount: 5
+    pollingInterval: 30
```
Comment (jacobwoffenden) [inline chart/values.yaml]: This can go as well

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

File: chart/values.yaml (hunk 8 above)
Comment (Gary-H9) [reply chart/values.yaml]: ✅

File: chart/values.yaml (hunk 9 above)
Comment (Gary-H9) [reply chart/values.yaml]: ✅

File: chart/values.yaml (hunk 6 above)
Comment (Gary-H9) [reply chart/values.yaml]: ✅

File: .github/workflows/codeql-analysis.yml (hunk 4 above)
Comment (Gary-H9) [reply .github/workflows/codeql-analysis.yml]: Updated across all workflows.

File: chart/values.yaml (hunk 7 above)
Comment (Gary-H9) [reply chart/values.yaml]: Tuned down, can be altered when built

**Review by jacobwoffenden — APPROVED** [review]

**Pull request merged**